### PR TITLE
CATROID-817 User defined Brick Parameters

### PR DESCRIPTION
**Pull request by devsebastian**
Ticket: https://jira.catrob.at/browse/CATROID-817

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [ ] Include the name of the Jira ticket in the PR’s title
- [ ] Include a summary of the changes plus the relevant context
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no compiler or linter warnings
- [ ] Perform a self-review of the changes
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s gitflow workflow
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by adnaribo — CHANGES_REQUESTED** [review]
Please check what is written under data in the formula editor for the values for the input of the user defined brick, see screenshot below. 
image
Even after executing the CTL of this ticket there are the IDs in the data field of the formula editor, see screenshot below.
image

File: catroid/src/main/java/org/catrobat/catroid/content/UserDefinedScript.java [hunk 1]
```diff
@@ -58,6 +61,19 @@ public EventId createEventId(Sprite sprite) {
...
+		this.userDefinedBrickInputs = userDefinedBrickInputs;
+	}
+
+	public UserData getUserDefinedBrickInput(String name) {
+		for (Object variable : userDefinedBrickInputs) {
+			if (variable instanceof UserData && ((UserData) (variable)).getName().equals(name)) {
+				return (UserData) variable;
+			}
+		}
+		return null;
```
[the remark this answers is not in the document]
Comment (mstoeg) [reply catroid/src/main/java/org/catrobat/catroid/content/UserDefinedScript.java:74]: In order to make the value part of the DataListAdapter display a value please change the Adapter for the User Defined Bricks from `UserDataRVAdapter` back to `UserDefinedBrickInputRVAdapter`. This class already exists, but somehow wasn't used any more. In the `UserDefinedBrickInputRVAdapter` class change the position where the value is extracted inside the `onBindViewHolder` method like this: 
```
	@Override
	public void onBindViewHolder(CheckableVH holder, int position) {
		super.onBindViewHolder(holder, position);

		UserDefinedBrickInput item = items.get(position);
		VariableVH variableVH = (VariableVH) holder;
		variableVH.title.setText(item.getName());
		int value;
		try {
			value = ((Formula) item.getValue()).interpretInteger(null);
		} catch (InterpretationException e) {
			value = 0;
		}
		variableVH.value.setText(toMetricUnitRepresentation(value));
	}
```

**Review by adnaribo — APPROVED** [review]
Looks good, thank you very much :1st_place_medal:

**Comment by wslany** [comment]
Thanks!!

**Pull request merged**